packages/ckeditor5-ui/src/highlightedtext/highlightedtextview.ts

```diff
@@ -0,0 +1,126 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module ui/highlightedtext/highlightedtextview
+ */
+
+import View from '../view';
+import { escape } from 'lodash-es';
+
+import '../../theme/components/highlightedtext/highlightedtext.css';
+
+/**
+ * A class representing a view that displays a text which subset can be highlighted using the
+ * {@link #highlightText} method.
+ */
+export default class HighlightedTextView extends View {
+	/**
+	 * The text that can be highlighted using the {@link #highlightText} method.
+	 *
+	 * **Note:** When this property changes, the previous highlighting is removed.
+	 */
+	declare public text: string | undefined;
+
+	/**
+	 * @inheritDoc
+	 */
+	constructor() {
+		super();
+
+		this.set( 'text', undefined );
+
+		this.setTemplate( {
+			tag: 'span',
+			attributes: {
+				class: [ 'ck', 'ck-highlighted-text' ]
+			}
+		} );
+
+		this.on( 'render', () => {
+			// Classic setTemplate binding for #text will not work because highlightText() replaces the
+			// pre-rendered DOM text node new a new one (and <mark> elements).
+			this.on( 'change:text', () => {
+				this._updateInnerHTML( this.text );
+			} );
+
+			this._updateInnerHTML( this.text );
+		} );
+	}
+
+	/**
+	 * Highlights view's {@link #text} according to the specified RegExp. If the passed RegExp is `null`, the
+	 * highlighting is removed
+	 *
+	 * @param regExp
+	 */
+	public highlightText( regExp: RegExp | null ): void {
+		this._updateInnerHTML( markText( this.text || '', regExp ) );
+	}
+
+	/**
+	 * Updates element's innerHTML with the passed content.
+	 *
+	 * @param newText
+	 */
+	private _updateInnerHTML( newInnerHTML: string | undefined ) {
+		this.element!.innerHTML = newInnerHTML || '';
+	}
+}
+
+// Replaces RegExp occurrences with <mark> tags in a text.
```

Comment:
Use `/** ... */` instead of `//`. This function is not exported, so it will not land in API docs. But it would enable hinting in IDE.